light-prover/prover/circuit_utils.go

```diff
@@ -222,38 +222,53 @@ func LoadVerifyingKey(filepath string) (verifyingKey groth16.VerifyingKey, err e
 
 	return verifyingKey, nil
 }
-
-func GetKeys(keysDir string, circuitTypes []CircuitType) []string {
+func GetKeys(keysDir string, circuitTypes []CircuitType, isLightweight bool) []string {
 	var keys []string
 
 	if IsCircuitEnabled(circuitTypes, Inclusion) {
-		keys = append(keys, keysDir+"inclusion_26_1.key")
-		keys = append(keys, keysDir+"inclusion_26_2.key")
-		keys = append(keys, keysDir+"inclusion_26_3.key")
-		keys = append(keys, keysDir+"inclusion_26_4.key")
-		keys = append(keys, keysDir+"inclusion_26_8.key")
+		if isLightweight {
+			keys = append(keys, keysDir+"inclusion_26_1.key")
+		} else {
+			keys = append(keys, keysDir+"inclusion_26_1.key")
+			keys = append(keys, keysDir+"inclusion_26_2.key")
+			keys = append(keys, keysDir+"inclusion_26_3.key")
+			keys = append(keys, keysDir+"inclusion_26_4.key")
+			keys = append(keys, keysDir+"inclusion_26_8.key")
+		}
 	}
 	if IsCircuitEnabled(circuitTypes, NonInclusion) {
-		keys = append(keys, keysDir+"non-inclusion_26_1.key")
-		keys = append(keys, keysDir+"non-inclusion_26_2.key")
+		if isLightweight {
+			keys = append(keys, keysDir+"non-inclusion_26_1.key")
+		} else {
+			keys = append(keys, keysDir+"non-inclusion_26_1.key")
+			keys = append(keys, keysDir+"non-inclusion_26_2.key")
+		}
 	}
 	if IsCircuitEnabled(circuitTypes, Combined) {
-		keys = append(keys, keysDir+"combined_26_1_1.key")
-		keys = append(keys, keysDir+"combined_26_1_2.key")
-		keys = append(keys, keysDir+"combined_26_2_1.key")
-		keys = append(keys, keysDir+"combined_26_2_2.key")
-		keys = append(keys, keysDir+"combined_26_3_1.key")
-		keys = append(keys, keysDir+"combined_26_3_2.key")
-		keys = append(keys, keysDir+"combined_26_4_1.key")
-		keys = append(keys, keysDir+"combined_26_4_2.key")
+		if isLightweight {
+			keys = append(keys, keysDir+"combined_26_1_1.key")
+		} else {
+			keys = append(keys, keysDir+"combined_26_1_1.key")
+			keys = append(keys, keysDir+"combined_26_1_2.key")
+			keys = append(keys, keysDir+"combined_26_2_1.key")
+			keys = append(keys, keysDir+"combined_26_2_2.key")
+			keys = append(keys, keysDir+"combined_26_3_1.key")
+			keys = append(keys, keysDir+"combined_26_3_2.key")
+			keys = append(keys, keysDir+"combined_26_4_1.key")
+			keys = append(keys, keysDir+"combined_26_4_2.key")
+		}
```

Comment:
this might brick program and js tests, maybe add another flag for program tests which downloads all of these?